httpcore5/src/test/java/org/apache/hc/core5/http/TestHttpHost.java

```diff
@@ -69,6 +73,27 @@ public void testConstructor() {
         Assert.assertThrows(NullPointerException.class, () -> new HttpHost(null, (String) null, -1));
         Assert.assertThrows(IllegalArgumentException.class, () -> new HttpHost(null, "   ", -1));
         Assert.assertThrows(NullPointerException.class, () -> new HttpHost(null, (InetAddress) null, -1));
+
+        final Host host = new Host("myHost", 8080);
```

Comment:
@arturobernalg Please add a test case with non-default scheme such as `https`. I suspect there will be an unpleasant surprise.

Reply:
HI @ok2c 
I know. but it the same  whit ->
`public HttpHost(final URIAuthority authority)`
What you suggest that we should do?

Reply:
> I know. but it the same whit the Authority constructor

@arturobernalg It is not the same. The URI authority does not have a scheme component.
> What you suggest that we should do?

I suggest that we do nothing. Not only this constructor is not very useful, it can lead to incorrect / unpredictable results.

Reply:
No, not nothing. I am going to deprecate `public HttpHost(final URIAuthority authority)`.

Reply:
OK. 
Close the PR:
TY